specs/agents/span-links.md

```diff
@@ -0,0 +1,36 @@
+## Span Links
+
+A Span or Transaction MAY be linked to zero or more other Spans/Transactions that are causally related.
+
+Specific use-cases for Span Links:
+
+1. When a single transaction represents the batch processing of several messages, the agent SHOULD be able to link back to the traces that have produced the messages.
+2.  When the agent receives a `traceparent` header from outside a trust boundary, it SHOULD restart the trace (creating a different trace id with its own sampling decision) and link to the originating trace.
+3. Close gap for the OTLP intake - [OTel's specification of span links](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#links-between-spans)
+
+Spans and Transactions MUST collect links in the `links` array with the following fields on each item:
+- `trace.id`: the id of the linked trace.
+- `span.id`: the id of the linked span or transaction.
+
+Example:
+
+```
+ "links": [
+      {"trace": {"id": "traceId1"}, "span": {"id": "spanId1"}},
+      {"trace": {"id": "traceId2"}, "span": {"id": "spanId2"}},
+    ]
+```
+
+## Trace Continuation
+
+We can expect incoming requests into an application with our agent which contains a `traceparent` header added by a 3rd party component. In this situation we end up with traces where part of the trace is outside of our system.
```

Comment:
```suggestion
The `traceparent` header of requests that are traces with our agents might have been added by a 3rd party component.
This situation becomes more and more common as the w3c trace context gets adopted.
If our agents blindly trust the `traceparent` header, we end up with traces where part of the trace is outside of our system and were we don't have control over the sampling rate.
```

Reply:
I think there are good ideas here (some I already took), 2 things I'd like to clarify:

1. You suggest 
> The `traceparent` header of requests that are traces with our agents

Are you sure about the "requests that are traces with our agent" part? I think it's hard to understand... is it maybe "requests that are traced with our agents"? 

2. 
> If our agents blindly trust the `traceparent` header, we end up with traces where part of the trace is outside of our system and were we don't have control over the sampling rate.

Do we really need to go into the sample rate here? I think currently there are some other issues when part of the trace is outside our system, so I'd not focus on specific issues. I'd just state that part of the trace is outside our system and this section of the spec intends to handle this in general.

Additionally the currently default `continue_always` strategy will allow traces where part of it is outside of our system which isn't reallyt aligned with saying "If our agents blindly trust the `traceparent` header, we end up with traces where part of the trace is outside of our system"

So overall in this part my intention was to say:
- there could be incoming requests with `tracestate` added by a 3rd party (defining what's the use case)
- this may lead to traces where part of the trace is outside our system (problem statement)
- the agent should be able to deal with this (and the way to deal with it is specified later)

I adapted this part, let me know if there is something left we should change.

Reply:
sounds good, thanks